snuba/cli/subscriptions_scheduler.py

```diff
@@ -56,11 +60,42 @@ def subscriptions_scheduler(
     schedule_ttl: int,
     log_level: Optional[str],
     delay_seconds: Optional[int],
-    load_factor: int = 1
+    load_factor: int = 1,
 ) -> None:
     """
-    Currently the subscriptions scheduler just consumes the commit log and records
-    the lag between the slowest and fastest partitions.
+    The subscriptions scheduler's job is to schedule subscriptions for a single entity.
+    It consumes the commit log for that entity which is used as a clock and determines
+    which subscriptions to run at each interval. It produces a message for each
+    scheduled subscription task to the scheduled subscription topic for that entity, so
+    it can be picked up and run by subscription executors.
+
+    The subscriptions scheduler consists of a tick consumer and three processing steps.
+
+    - The tick consumer consumes the commit log and reads the "orig_message_ts" header.
+    It constructs a new `Tick` message representing the intervals between each of the
+    original messages, which gets passed to the processing strategy. Note: A tick always
+    corresponds to a single partition on the original topic (not the commit log topic
+    as that is never partitioned).
+
+    - The first processing step is a tick buffer. It buffers ticks where needed and
+    determines when to submit them to the rest of the pipeline. The tick buffer behavior
+    depends on the watermark mode specsified by the entity. In PARTITION mode, ticks are
+    never buffered and immediately submitted to the next step. In GLOBAL mode we wait
+    (filling the buffer) until the timestamp of a tick has been reached on every
+    partition before eventually submitting a tick to the next step. This guarantees that
+    a subscription is never be scheduled before data on every partition up to that
+    timestamp is written to storage.
+
+    - The second processing step provides the strategy for committing offsets. Ticks are
+    only marked as `should_commit` if every partition has already reached the timestamp
+    of the tick. Only the offset of the slowest partition will get committed. This
+    guarantees at least once scheduling of subscriptions.
```

Comment:
You may want to clarify that it is only the offset "on the commit log" of the slowest partition "on the main topic" that gets committed.